### ISSUE TITLE
Support for elegoo mars 2 pro

### ISSUE THREAD

**Issue by helmet86** [Andrew-Dickinson/photonic-etcher#2]
Hello,

This Project is what I have been looking for weeks now.
Is there a way to support elegoo printers or export to stl file for bigger support on other printers ?

**Comment by t2rtech** [comment]
Great Project!! Congratulations, and thanks for sharing!
+1 for STL Support, allowing use with third-party slicers like Chitubox

**Comment by AABatteries** [comment]
Interested in support for Elegoo Mars 3 as well, so Chitu-based printers and/or .stl export support for a more generic approach :)

**Comment by Andrew-Dickinson** [comment]
Hey folks, `STL` support is something I considered. However, I think the existing tooling is pretty close for getting you there. If you open Gerber files with Fusion 360, you can render them into 3D models and export to STL relatively easily. Thomas Sanladerer demonstrates this technique in [his video](https://www.youtube.com/watch?v=RudStbSApdE) on this process. 

My goal with this project was a "one click" workflow for going straight from gerber to printable file, which I have achieved for the Anycubic printers (or I at least I hope to shortly). I'm happy to consider other printers if there's demand for it.

What is the file extension used by the the Elegoo Mars printers?

**Comment by daxliniere** [comment]
Yes, I agree Andrew, STL support seems like a step away from the underlying concept. Going from a Gerber file to a 3D model to a flat/sliced file is a bit backwards.

**Comment by helmet86** [comment]
I tried the process from Thomas to convert Gerber > SVG > Fusion STL.
But the particular file generated from KiCad somehow was garbage already converted to SVG.
After import to Fusion 360 the sizes are not correct anymore and pads are incomplete.
Also tried to use different software i could find to convert Gerber to SVG and nothing did the job right.

So if your tool could also support Elegoo printers would be very helpfull to remove the errors introduced due to rescaling etc.

Mars 2 Pro details from chitubox and attached also file example with 3D-benchy.
Format is .CTB

![image](https://user-images.githubusercontent.com/49732558/184529082-059ca223-4cad-48e6-b6bb-0cec68940f9f.png)
[3DBenchy_CTB.zip](https://github.com/Andrew-Dickinson/photonic-etcher/files/9332964/3DBenchy_CTB.zip)

**Comment by Andrew-Dickinson** [comment]
@helmet86 Does the board preview in photonic-etcher look correct for your boards? I've never tried using KiCad gerbers, maybe they have a different encoding? 

Photonic-etcher uses a third-party JS library called [tracespace](https://github.com/tracespace/tracespace) to identify gerber layers, render board previews, and convert to SVG. I suspect tracespace handles any peculiarities of the KiCad gerber exports, but like I said, I've never tried it.

**Comment by Kobaloi** [comment]
Love this idea.  And great work getting it to work on the anycubic machines. I'd also love to do this with the Elegoo machines.

**Comment by Coffee2CodeNL** [comment]
+1 for Elegoo

**Comment by github-1a** [comment]
+1 for Elegoo

Elegoo printers are truly wide spread, especially MARS 3 and SATURN are great for this purpose. Its really a great "one click" way. Rather I will not tell you how awkward conversion I still use for my Saturn.

**Comment by melx2** [comment]
+1 for Elegoo

Support for Chitubox files would be great. I've tried the Thomas Sanladerer way and like others I found there were issues subtracting the traces to create the stencil. Your one click approach looks good.

**Comment by dsmaddalena** [comment]
The problem for the Saturn through Mars 4 DLP is the .ctb format, which requires requesting the SDK from Chitu Systems to break through the encryption. Elegoo has moved on to .GOO format starting the the Saturn 3/Mars 4 6k, so proper open source. The positive about opening it up to .ctb would be possible coverage for quite a few brands that were bound to the encrypted Chitu boards... Phrozen 4ks and onwards are using them. It's about 3 years worth of consumer printers.

**Comment by HappyTechnician** [comment]
PLEASE. I would like to convert to Chitubox slicer format. I have a Creality-006 resin printer and I would love to make PCBs with it. I would even use .stl files if possible...
I am currently using Fusion 360 for my designs... Thank You so much!!!
